feat: add `number/uint32/base/identity`

gunjjoshi opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: passed
  - task: lint_c_examples status: passed
  - task: lint_c_benchmarks status: passed
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves None.

## Description

> What is the purpose of this pull request?

This pull request:

-   adds `number/uint32/base/identity`.

## Related Issues

> Does this pull request have any related issues?

None. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

gunjjoshi committed:
feat: add number/uint32/base/identity

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/number/uint32/base/identity/README.md:
```diff
@@ -0,0 +1,214 @@
+    'dtype': 'uint32'
+};
+
+var x = discreteUniform( 100, 0, 50, opts );
```
@gunjjoshi Would you mind refactoring this example to use `logEachMap`, similar to https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/number/float64/base/add#examples? We've been trying to move away from manual `for` loops.

gunjjoshi replied:
Updated the JS examples, thanks!

kgryte commented on lib/node_modules/@stdlib/number/uint32/base/identity/benchmark/benchmark.native.js:
```diff
@@ -0,0 +1,63 @@
+
+// MAIN //
+
+bench( pkg, opts, function benchmark( b ) {
```
```suggestion
bench( pkg+'::native', opts, function benchmark( b ) {
```

kgryte committed:
bench: fix missing namespace

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/number/uint32/base/identity/benchmark/c/native/benchmark.c:
```diff
@@ -0,0 +1,137 @@
+	elapsed = tic() - t;
+	if ( y != y ) {
+		printf( "should not return NaN\n" );
+	}
```
```suggestion
		if ( y > 100 ) {
			printf( "unexpected result\n" );
			break;
		}
	}
	elapsed = tic() - t;
	if ( y > 100 ) {
		printf( "unexpected result\n" );
	}
```

A good compiler will  recognize that a `NaN` check does not apply here.

kgryte commented on lib/node_modules/@stdlib/number/uint32/base/identity/benchmark/c/native/benchmark.c:
```diff
@@ -0,0 +1,137 @@
+static double benchmark( void ) {
+	uint32_t x[ 100 ];
+	double elapsed;
+	double y;
```
```suggestion
	uint32_t y;
```

kgryte commented on lib/node_modules/@stdlib/number/uint32/base/identity/examples/index.js:
```diff
@@ -0,0 +1,35 @@
+	'dtype': 'uint32'
+};
+
+var x = discreteUniform( 100, 0, 50, opts );
```
Same comments as README.

kgryte commented on lib/node_modules/@stdlib/number/uint32/base/identity/lib/main.js:
```diff
@@ -0,0 +1,50 @@
+* Evaluates the identity function for an unsigned 32-bit integer `x`.
+*
+* @param {integer} x - input value
+* @returns {integer} input value
```
```suggestion
* @param {uinteger} x - input value
* @returns {uinteger} input value
```

kgryte committed:
docs: update types

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/number/uint32/base/identity/lib/native.js:
```diff
@@ -0,0 +1,58 @@
+*
+* @private
+* @param {integer} x - input value
+* @returns {integer} input value
```
```suggestion
* @param {uinteger} x - input value
* @returns {uinteger} input value
```

kgryte committed:
docs: update types

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
bench: fix type

Signed-off-by: Athan <kgryte@gmail.com>

kgryte committed:
bench: update conditions

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (changes requested):
Other than the examples, this is looking good.

gunjjoshi committed:
docs: update js examples to use logEach

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM
